stats.cpp

```diff
@@ -1,5 +1,51 @@
 #include "stats.h"
 
-Stats Statistics::ComputeStatistics(const std::vector<___>& ) {
-    //Implement statistics here
+Stats::Stats(): average(NAN), max(NAN), min(NAN) { }
+
+Stats Statistics::ComputeStatistics(const std::vector<float>& numbers) {
+    Stats result;
+
+    if (numbers.empty())
+        return result;
+    else
+    {
+        result.average = Statistics::Average(numbers);
+        result.max = Statistics::Max(numbers);
+        result.min = Statistics::Min(numbers);
+    }
+    return result;
+}
+
+float Statistics::Average(const std::vector<float>& numbers) {
+    
+    float sum = 0;
+    
+    for (int i = 0; i < numbers.size(); i++)
+    { sum += numbers[i]; }
+
+    float average = sum / numbers.size();
+
+    return average;
+}
+
+float Statistics::Max(const std::vector<float>& numbers)
+{
+    float max = 0;
```

Comment:
Same comment as min (see below)... this will not work if all numbers are negative.